envoy/datadog_checks/envoy/parser.py

```diff
@@ -41,6 +41,10 @@ def parse_metric(metric, metric_mapping=METRIC_TREE):
         if metric_part in metric_mapping and num_tags >= minimum_tag_length:
             # Rebuild any built up tags whenever we encounter a known metric part.
             if tag_builder:
+                # Edge case where we hit a known metric part after a sequence of all unknown parts
+                if '|_tags_|' not in metric_mapping:
+                    raise UnknownMetric
```

Comment:
I thought this was not correct but seems that the PEP to make it incorrect got rejected https://www.python.org/dev/peps/pep-0317 :p